rllib/core/rl_module/torch/torch_rl_module.py

```diff
@@ -4,6 +4,12 @@
 from ray.rllib.core.rl_module import RLModule
 
 torch, nn = try_import_torch()
+if torch:
+    from torch.nn.parallel import DistributedDataParallel as DDP
+else:
+    raise RuntimeError(
```

Comment:
does this actually work when you run it in an environment without torch?
like there will not be DDP, and the class definition below will error out?

Reply:
Yes. I just verified it using this script on a conda env without torch installed. 

```
import os
import logging 
logger = logging.getLogger(__name__)

def try_import_torch(error: bool = False):
    """Tries importing torch and returns the module (or None).

    Args:
        error: Whether to raise an error if torch cannot be imported.

    Returns:
        Tuple consisting of the torch- AND torch.nn modules.

    Raises:
        ImportError: If error=True and PyTorch is not installed.
    """
    if "RLLIB_TEST_NO_TORCH_IMPORT" in os.environ:
        logger.warning("Not importing PyTorch for test purposes.")
        return _torch_stubs()

    try:
        import torch
        import torch.nn as nn

        return torch, nn
    except ImportError:
        if error:
            raise ImportError(
                "Could not import PyTorch! RLlib requires you to "
                "install at least one deep-learning framework: "
                "`pip install [torch|tensorflow|jax]`."
            )
        return _torch_stubs()


def _torch_stubs():
    nn = _NNStub()
    return None, nn


# Fake module for torch.nn.
class _NNStub:
    def __init__(self, *a, **kw):
        # Fake nn.functional module within torch.nn.
        self.functional = None
        self.Module = _ModuleStub


# Fake class for torch.nn.Module to allow it to be inherited from.
class _ModuleStub:
    def __init__(self, *a, **kw):
        raise ImportError("Could not import `torch`.")



torch, nn = try_import_torch()
if torch:
    from torch.nn.parallel import DistributedDataParallel as DDP
else:
    raise RuntimeError(
        "Torch is not installed. Please install torch or do pip install ray[rllib]."
    )


class Foo(DDP):
    def __init__(self):
        pass
```

To run it you can do `python -m test.py`.
It will raise the Runtime error.

Reply:
right, it will raise runtime error. but do you want it to raise runtime error was what I meant.
like there are TF users that don't have torch installed, and they just want to use RLlib TF2.
we shouldn't raise Runtime error for these users.
to be more specifically, this conditional import is something new that I haven't seen in our codebase. so a bit suspicous.
I would suggest we set DDP to something else in load time, but then will throw errors if someone actually tries to use it without torch.

Reply:
discussed offline, I will inherit from `nn.parallel.DistributedDataParallel`

Reply:
Fixed.